mypy/treetransform.py

```diff
@@ -285,7 +285,7 @@ def visit_with_stmt(self, node: WithStmt) -> WithStmt:
         return WithStmt(self.expressions(node.expr),
                         self.optional_expressions(node.target),
                         self.block(node.body),
-                        self.optional_type(node.target_type))
+                        self.optional_type(node.unanalyzed_type))
```

Comment:
Interestingly, we do the opposite for assignments, i.e. copy only the analyzed type. I think that we should copy both analyzed and unanalyzed types both here and for assignments.